app.py

```diff
@@ -6,7 +6,35 @@
 
 @app.route('/')
 def index():
-    return 'Hello World!'
+    #populate variable with the list of saved todo items
+    items = session.get_items()
+    #sort the list ased on the value of the status field in reverse order
+    items=sorted(items, key=lambda k: k['status'], reverse=True)
+    #pass the sorted list into the index template for display
+    return render_template('index.html', items=items)
+
+#route to allow updating of status for each item based on id number
+@app.route('/<id>', methods=['POST'])
```

Comment:
It would be a nicer if the route is clear about what it does.

It is not intuitive that a POST to /1 should switch the status of item with id 1. This structure makes it difficult to add new entities/update other parts of an item.
It is also not intuitive that a second POST request to /1 has the opposite effect - People could accidentally set the wrong status if they are viewing a page that is out of date. 

Other possibilities would be two routes:
/item/<id>/complete and /item/<id>/not-started

Or /item/<id> and the request includes a parameter for the new status.

Reply:
Thanks - this makes a lot of sense, especially after reading module 2